crates/ra_cargo_watch/src/lib.rs

```diff
@@ -67,22 +67,6 @@ impl CheckWatcher {
     }
 }
 
-impl std::ops::Drop for CheckWatcher {
-    fn drop(&mut self) {
-        if let Some(handle) = self.handle.take() {
-            // Take the sender out of the option
-            let cmd_send = self.cmd_send.take();
-
-            // Dropping the sender finishes the thread loop
-            drop(cmd_send);
```

Comment:
I believe the order of drop is crucial here, the `cmd_send` has to be dropped first such that the loop reading the cargo output terminates abruptly

Reply:
yes, the drop order of fields is guaranteed in Rust.